pkg/api/v1alpha1/versionskew.go

```diff
@@ -0,0 +1,37 @@
+package v1alpha1
+
+import (
+	"fmt"
+	"math"
+
+	"k8s.io/apimachinery/pkg/util/version"
+)
+
+const supportedMinorVersionIncrement = 1
+
+// ValidateVersionSkew validates Kubernetes version skew between valid non-nil versions.
+func ValidateVersionSkew(oldVersion, newVersion *version.Version) error {
+	newVersionMajor := newVersion.Major()
+	newVersionMinor := newVersion.Minor()
+	oldVersionMajor := oldVersion.Major()
+	oldVersionMinor := oldVersion.Minor()
+
+	majorVersionDifference := math.Abs(float64(newVersionMajor) - float64(oldVersionMajor))
```

Comment:
there is probably something I'm missing here, but why do we need to do a float subtraction to then abs it? major versions are already ints

Reply:
versions are actually uint. units type can not have a value of less than 0. Since we want to compare all the versions and check for conditions of negative and positive difference, we convert them to float64.